Libraries/Lists/VirtualizedSectionList.js

```diff
@@ -523,22 +515,28 @@ class ItemWithSeparator extends React.Component<
   }
 }
 
-function getItem(sections: ?$ReadOnlyArray<Item>, index: number): ?Item {
+function getItem(
+  props: Props<SectionBase>,
+  sections: ?$ReadOnlyArray<Item>,
+  index: number,
+): ?Item {
   if (!sections) {
     return null;
   }
   let itemIdx = index - 1;
-  for (let ii = 0; ii < sections.length; ii++) {
-    if (itemIdx === -1 || itemIdx === sections[ii].data.length) {
+  for (let ii = 0; ii < props.getItemCount(sections); ii++) {
```

Comment:
Working on this at FB. I think this is supposed to stay `sections.length` here because `sections` is always an array. I'm changing this internally, no need to do anything, just wanted to put this comment here for your information.

Reply:
@cpojer Thanks ! In fact it can be something else, by exemple if you use an immutable list, to get his length the method is `size` and not `length` so it's better to keep it outside to handle immutable data